From e3f1f25fa0344e4a981e8f5e5d5c3457ed420d35 Mon Sep 17 00:00:00 2001
From: Mark Mentovai <mark@moxienet.com>
Date: Wed, 17 Nov 2010 16:34:37 -0500
Subject: [PATCH 617/984] cfg80211: fix can_beacon_sec_chan, reenable HT40

[ upstream commit 09a02fdb919876c01e8f05960750a418b3f7fa48 ]

This follows wireless-testing 9236d838c920e90708570d9bbd7bb82d30a38130
("cfg80211: fix extension channel checks to initiate communication") and
fixes accidental case fall-through. Without this fix, HT40 is entirely
blocked.

Signed-off-by: Mark Mentovai <mark@moxienet.com>
Signed-off-by: Andi Kleen <ak@linux.intel.com>
Cc: stable@kernel.org
Acked-by: Luis R. Rodriguez <lrodriguez@atheros.com
Signed-off-by: John W. Linville <linville@tuxdriver.com>
---
 net/wireless/chan.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/net/wireless/chan.c b/net/wireless/chan.c
index 93d89d6..bbaab4b 100644
--- a/net/wireless/chan.c
+++ b/net/wireless/chan.c
@@ -53,8 +53,10 @@ static bool can_beacon_sec_chan(struct wiphy *wiphy,
 	switch (channel_type) {
 	case NL80211_CHAN_HT40PLUS:
 		diff = 20;
+		break;
 	case NL80211_CHAN_HT40MINUS:
 		diff = -20;
+		break;
 	default:
 		return false;
 	}
-- 
1.8.1.5

